lib/network/graph.py

```diff
@@ -0,0 +1,55 @@
+import networkx as nx
+
+class BaseGraph():
+
+    def __init__(self):
+        # to do here
+        pass
+
+    def graph_add_node(self, n, attr_dict=None, **attr):
```

Comment:
@bhuvangupta008 This is a very transparent wrapper. It does not provide any higher level abstraction that would allow us to replace networkx some day.

What I suggest is having general methods like - 
```
add_node()
remove_node()
add_edge()
remove_edge()
```
These methods can call the networkx methods for now, but may be substituted later.

Reply:
If needed we can have a class abstraction for node and edge. Both nodes and edges shall have properties which will become important later on. You can think about methods for adding and removing properties of nodes and edges.